src/translator/parser.cpp

```diff
@@ -0,0 +1,164 @@
+  app.add_flag("--build-info", build_info_, "Print build-info and exit");
+  app.add_flag("--version", version_, "Print version-info and exit");
+}
+
+void ConfigParser::handleSpecialOptions() {
+  if (build_info_) {
+#ifndef _MSC_VER  // cmake build options are not available on MSVC based build.
+    std::cerr << cmakeBuildOptionsAdvanced() << std::endl;
+    exit(0);
+#else   // _MSC_VER
+    ABORT("build-info is not available on MSVC based build.");
+#endif  // _MSC_VER
+  }
+
+  if (version_) {
+    std::cerr << buildVersion() << std::endl;
+    exit(0);
+  }
+}
+
+void ConfigParser::addOptionsBoundToConfig(CLI::App &app, CLIConfig &config) {
+  app.add_option("--model-config-paths", config.modelConfigPaths,
+                 "Configuration files list, can be used for pivoting multiple models or multiple model workflows");
+
+  app.add_flag("--bytearray", config.byteArray,
+               "Flag holds whether to construct service from bytearrays, only for testing purpose");
+
+  app.add_flag("--check-bytearray", config.validateByteArray,
+               "Flag holds whether to check the content of the bytearrays (true by default)");
+
+  app.add_option("--cpu-threads", config.numWorkers, "Number of worker threads to use for translation");
+
+  app_.add_option("--bergamot-mode", config.opMode, "Operating mode for bergamot: [wasm, native, decoder]");
+}
+
+std::shared_ptr<marian::Options> parseOptionsFromFilePath(const std::string &configPath, bool validate /*= true*/) {
+  // Read entire string and redirect to parseOptionsFromString
+  std::ifstream readStream(configPath);
+  std::stringstream buffer;
+  buffer << readStream.rdbuf();
+  return parseOptionsFromString(buffer.str(), validate, /*pathsInSameDirAs=*/configPath);
+};
+
+std::shared_ptr<marian::Options> parseOptionsFromString(const std::string &configAsString, bool validate /*= true*/,
+                                                        std::string pathsInSameDirAs /*=""*/) {
+  marian::Options options;
+
+  marian::ConfigParser configParser(cli::mode::translation);
+
+  // These are additional options we use to hijack for our own marian-replacement layer (for batching,
+  // multi-request-compile etc) and hence goes into Ptr<Options>.
+  configParser.addOption<size_t>("--max-length-break", "Bergamot Options",
+                                 "Maximum input tokens to be processed in a single sentence.", 128);
+
+  // The following is a complete hijack of an existing option, so no need to add explicitly.
+  // configParser.addOption<size_t>("--mini-batch-words", "Bergamot Options",
+  //                                "Maximum input tokens to be processed in a single sentence.", 1024);
+
+  configParser.addOption<std::string>("--ssplit-prefix-file", "Bergamot Options",
+                                      "File with nonbreaking prefixes for sentence splitting.");
+
+  configParser.addOption<std::string>("--ssplit-mode", "Bergamot Options", "[paragraph, sentence, wrapped_text]");
```

Comment:
@jerinphilip The default value for `--ssplit-mode` option has been removed. Is it intended?

Reply:
Not really. Will set back in a follow-up commit.